matrix_openCL/Project1/main.cpp

```diff
@@ -0,0 +1,88 @@
+#define __CL_ENABLE_EXCEPTIONS
+#include <CL/cl.hpp>
+#include <utility>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <vector>
+#include <array>
+using namespace std;
+using namespace cl;
+
+int main() {
+    // Create the two input vectors
+    const int LIST_SIZE = 1000;
+    array<int, LIST_SIZE> A = {};
+    array<int, LIST_SIZE> B = {};
+    for (int i = 0; i < LIST_SIZE; i++) {
+        A[i] = i;
+        B[i] = LIST_SIZE - i;
+    }
+
+    try {
+        // Get available platforms
+        vector<Platform> platforms;
+        Platform::get(&platforms);
+
+        // Select the default platform and create a context using this platform and the GPU
+        cl_context_properties cps[3] = {
+            CL_CONTEXT_PLATFORM,
+            (cl_context_properties)(platforms[0])(),
+            0
+        };
+        Context context(CL_DEVICE_TYPE_GPU, cps);
+
+        // Get a list of devices on this platform
+        vector<Device> devices = context.getInfo<CL_CONTEXT_DEVICES>();
+
+        // Create a command queue and use the first device
+        CommandQueue queue = CommandQueue(context, devices[0]);
+
+        // Read source file
+        std::ifstream sourceFile("vector_adds_kernel.cl");
+        std::string sourceCode(
+            std::istreambuf_iterator<char>(sourceFile),
+            (std::istreambuf_iterator<char>()));
+        Program::Sources source(1, std::make_pair(sourceCode.c_str(), sourceCode.length() + 1));
+
+        // Make program of the source code in the context
+        Program program = Program(context, source);
+
+        // Build program for these specific devices
+        program.build(devices);
+
+        // Make kernel
+        Kernel kernel(program, "vector_add");
+
+        // Create memory buffers
+        Buffer bufferA = Buffer(context, CL_MEM_READ_ONLY, LIST_SIZE * sizeof(int));
+        Buffer bufferB = Buffer(context, CL_MEM_READ_ONLY, LIST_SIZE * sizeof(int));
+        Buffer bufferC = Buffer(context, CL_MEM_WRITE_ONLY, LIST_SIZE * sizeof(int));
+
+        // Copy lists A and B to the memory buffers
+        queue.enqueueWriteBuffer(bufferA, CL_TRUE, 0, LIST_SIZE * sizeof(int), &A[0]);
+        queue.enqueueWriteBuffer(bufferB, CL_TRUE, 0, LIST_SIZE * sizeof(int), &B[0]);
+
+        // Set arguments to kernel
+        kernel.setArg(0, bufferA);
+        kernel.setArg(1, bufferB);
+        kernel.setArg(2, bufferC);
+
+        // Run the kernel on specific ND range
+        NDRange global(LIST_SIZE);
+        NDRange local(1);
```

Comment:
What is this? if i change the parameter value to small number like 20 or 100, it doesn't change anything, the result is still ok. But at bigger numbers, like 200 or more it gives an error. I don't know the exact value where it starts giving an error.

Reply:
Have you searched what that parameter is for? why would you need to change it?

Reply:
a quick google search tells that this is the number of work items in a work group. a work group is used for synchronisation purposes. you don't need to modify it here